src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,235 @@
+        } catch (DateTimeException | NumberFormatException e) {
+            throw new InvalidFormatDateException();
+        }
+    }
+
+    /**
+     * Making sense of the user's input and outputting the relevant commands to the input.
+     *
+     * @param message The inputted user's command.
+     * @return A command object that executes the user's command.
+     * @throws InvalidFormatByeException Throws an InvalidFormatByeException when the format of Bye is incorrect.
+     * @throws InvalidFormatListException Throws an InvalidFormatListException when the format of List is incorrect.
+     * @throws InvalidFormatDoneException Throws an InvalidFormatDoneException when the format of Done is incorrect.
+     * @throws EmptyTextException Throws an EmptyTextException when the user did not specify anything after the command.
+     * @throws InvalidFormatDeleteException Throws an InvalidFormatDeleteException when the format of Delete is
+     * incorrect.
+     */
+    public static Command parse(String message) throws InvalidFormatByeException, InvalidFormatListException,
+            InvalidFormatDoneException, EmptyTextException, InvalidFormatDeleteException, InvalidFormatFindException {
+        String[] inputArr = message.trim().replaceAll("  +", " ").split(" ", 2);
+        inputArr[0] = inputArr[0].toLowerCase();
+        if (isBye(inputArr[0])) {
+            if (inputArr.length != 1) {
+                throw new InvalidFormatByeException();
+            }
+            return new ByeCommand(inputArr);
+        } else if (isList(inputArr[0])) {
+            if (inputArr.length != 1) {
+                throw new InvalidFormatListException();
+            }
+            return new ListCommand(inputArr);
+        } else if (isValidDone(inputArr[0])) {
+            // checking if the input is valid
+            if (inputArr.length == 1 || !isNumber(inputArr[1])) {
+                throw new InvalidFormatDoneException();
+            }
+            return new DoneCommand(inputArr);
+        } else if (isTask(inputArr[0])) {
+            // checking if the input is valid
+            if (inputArr.length == 1) {
+                throw new EmptyTextException(inputArr[0]);
+            }
+            return new AddCommand(inputArr);
+        } else if (isDelete(inputArr[0])) {
+            // checking if the input is valid
+            if (inputArr.length == 1) {
+                throw new InvalidFormatDeleteException();
+            }
+            return new DeleteCommand(inputArr);
+        } else if (isFind(inputArr[0])) {
+            if (inputArr.length == 1) {
+                throw new InvalidFormatFindException();
+            }
+            String[] inputArr2 = inputArr[1].split(" ");
+            if (inputArr2.length > 1) {
+                throw new InvalidFormatFindException();
+            }
+            return new FindCommand(inputArr);
+        } else {
+            return new UnknownCommand(inputArr);
+        }
+    }
+}
```

Comment:
You can try using switch case to compare keywords here! It will help to reduce your code and make it look neater too. Also with switch case, you can omit the methods such as isDelete, isFind... etc. 
Is it also possible to bring out some of the if statements too? Because I see many recurring if statements in the method.